portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -51,26 +63,68 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      String comment = request.getParameter("comment");
+    String comment = request.getParameter("comment");
+    String imageUrl = getUploadedFileUrl(request, "comment-image");
       if(!comment.trim().equals("")) {
           Entity commentEntity = new Entity("Comment");
           commentEntity.setProperty("message", comment);
           commentEntity.setProperty("time", System.currentTimeMillis());
+          commentEntity.setProperty("imageUrl", imageUrl);
           DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
           datastore.put(commentEntity);
       }
       response.setContentType("text/html");
       response.sendRedirect("/index.html");
   }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
```

Comment:
I don't understand what (dev server) means here.  Do you expect that this could happen in dev, but not in live?  Why is that?

Probably add something to this comment like "This could happen on the dev server because blah, blah, blah."

Reply:
Basically, try to comment like you're going to be looking at this code in three years and trying to figure out what you were thinking when you wrote it.

Reply:
This method is from one of the files given in the walkthrough, I think that the generated URL has a different format when running on the "development server" vs the "live server".